### ISSUE TITLE
Set large_file_sha1 for large files.

### ISSUE THREAD

**Issue by bwbeach** [Backblaze/B2_Command_Line_Tool#284]
The B2 docs recommend the file info `large_file_sha1` for storing the checksum of the entire contents of a large file.  It would be nice for the command-line tool to know how to set it.  The drawback to setting it is that it requires reading the entire file to compute the sha1 before starting to upload it.

The question is: Should setting it be the default (with an option to turn it off), or should setting it require an explicit command-line option?

**Comment by ppolewicz** [comment]
What benefit (other than following the documentation) for the end user would that achieve?

IMO hashes of parts are sufficient and an overall hash is not worth the additional read or client-side implementation of it.

Enabling this by default could severely impact some of our users by consuming additional resources and failing to fit in the set backup window. I think we should keep the compatibility in this case.

**Comment by svonohr** [comment]
I agree with @ppolewicz. There is really not much you can do with these hashes, since you can't rely on them being available (or even correct). But I think it would be useful to have an optional flag for this, if the user needs the hash for a specific scenario.

Also, it would be easier on the client if the hash can be specified after uploading. For large file hashes they could be supplied to the `b2_finish_large_file function`.

**Comment by bwbeach** [comment]
I agree that this should be an option you can enable, not the default.

@svonohr: Eventually, we want to be able to supply this checksum at the end.  Even that, thought, will require processing the parts in sequence, not in parallel, to compute the checksum.

**Comment by ppolewicz** [comment]
If `--threads 1` is used or if the sync planner is smart, providing the hash at the end can avoid the additional read.

**Comment by eonil** [comment]
I think one important thing is missing here.

We upload files *to use them later*. Having no error on server-side is actually only half-work and not really helpful to archive integrity when I downloaded the file. Integrity scenario should include the case after the file after downloaded.

After downloading, the only way to verify integrity of downloaded file is its SHA1 hash. If I don't have a SHA1 hash, there's no way to verify whether the download file is fine or damaged. (please correct me if I'm wrong!) Therefore, I think Backblaze should *require* SHA1 for all uploaded file. Missing SHA1 should be treated as *incomplete*.

IMO, with that in mind, option `--sha1` should become an overriding switch rather than an optional attachment. Users want `b2` command would calculate SHA1 automatically if they consider "full integrity" and `--sha1` overriding has not been provided. 

> If you do not provide SHA1 (regardless of verified by server or your own provision), your upload is incomplete.

With that in mind, I can imagine a command line switch that calculates SHA1 automatically. Such as `b2 upload-file --calculate-sha1-automatically`. Also command line manual should clearly mention that the switch does not make server to verify the whole content SHA1.

---------

If `b2` command line cannot accept behavioral modification, I think you can provide extra switch like `--autocalc-sha1`.

-----------

Or Backblaze can provide access to original uploading segments. (if it exists...) I don't know how files are stored on server-side, but if Backblaze keeps original segments and their SHA1 hashes, everything is done simple and beautiful and nothing is really required. Segmented download is preferred even for client side. (Please consider that web is not the only client)

**Comment by gcstang** [comment]
How do you get the sha1 back when performing a download of a previously large multipart upload?
The X-Bz-Content-Sha1 field itself is empty and I'm not understanding how to use the b2ContentDisposition field during the download of such a file.
